src/sagas.js

```diff
@@ -457,22 +457,26 @@ function* startProgress ({ vmId, poolId, name }) {
   }
 }
 
+function* getSingleInstance ({ result, poolId }) {
+  yield fetchSingleVm(getSingleVm({ vmId: result.vm.id }))
+  if (poolId) {
+    yield fetchSinglePool(getSinglePool({ poolId }))
+  }
+}
+
 function* stopProgress ({ vmId, poolId, name, result }) {
-  const fetchSingle = vmId ? fetchSingleVm : fetchSinglePool
-  const getSingle = vmId ? getSingleVm : getSinglePool
   const actionInProgress = vmId ? vmActionInProgress : poolActionInProgress
-
-  const params = vmId ? { vmId } : { poolId }
   if (result && result.status === 'complete') {
     // do not call 'end of in progress' if successful,
     // since UI will be updated by refresh
     yield delay(5 * 1000)
-    yield fetchSingle(getSingle(params))
+    yield getSingleInstance({ result, poolId })
```

Comment:
Can we pass `vmId` instead of the `result` here?

Reply:
Done.